### ISSUE TITLE
Allow a node to require extend

### ISSUE THREAD

**Issue by triwav** [TwitchBronBron/vscode-brightscript-language#142]
So just learned that a node doesn't have to extend from another node type as it appears to fall back to using Group by default. This should not be reported as an error like it currently is `Component must have an extends attribute.brs(1007)`

**Comment by TwitchBronBron** [comment]
Very interesting! So then, do you see any value in identifying this ever? I could downgrade this to a warning, (which can be turned off by using `ignoreErrorCodes`)....or I could just remove the check entirely. Thoughts?

**Comment by triwav** [comment]
Yeah I'm cool with a warning especially if we explain why it still works.
